### PR TITLE
[JSDoc] Add missing documentation for botframework-streaming

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 2602

## Description
This PR introduces the change to `jsdoc/require-jsdoc` in the [.eslintrc.json](https://github.com/microsoft/botbuilder-js/blob/master/.eslintrc.json#L263) file from `warn` to `error` and adds all the missing JSDoc documentation for [botframework-streaming](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botframework-streaming) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

> **Note:** We divided the work for this library into two PRs to ease the review. PR# 151 contains the rest of the botbuilder fixes.

## Specific Changes

  - Updated `jsdoc/require-jsdoc` in `.eslintrc.json` from `warn` to `error`.
  - Private methods' documentation that does not exist on botbuilder-dotnet we added the `@private` tag in the JSDoc comment.
  - Added JSDoc comments in **16 files**.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

File: libraries/botframework-streaming/src/assemblers/payloadAssembler.ts [hunk 1]
```diff
@@ -46,6 +51,10 @@ export class PayloadAssembler {
         this._onCompleted = params.onCompleted;
     }
 
+    /**
+     * Retrieves the assembler's payload as a stream.
+     * @returns >A `SubscribableStream` of the assembler's payload.
```
Comment (iscaltritti-sw) [inline libraries/botframework-streaming/src/assemblers/payloadAssembler.ts:56]:
```suggestion
     * @returns A `SubscribableStream` of the assembler's payload.
```

File: libraries/botframework-streaming/src/protocolAdapter.ts [hunk 2]
```diff
@@ -47,22 +50,27 @@ export class ProtocolAdapter {
     }
 
     /// <summary>
-    /// Sends a request over the attached request manager.
+    ///
     /// </summary>
     /// <param name="request">The outgoing request to send.</param>
     /// <param name="cancellationToken">Optional cancellation token.</param>
```
Comment (iscaltritti-sw) [inline libraries/botframework-streaming/src/protocolAdapter.ts:56]: The whole comment block should be removed?

Comment (sw-joelmut) [reply libraries/botframework-streaming/src/protocolAdapter.ts]: Yea, the comment block should be removed. Thank you, I missed that.

**Review by santgr11 — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Review by iscaltritti-sw — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**